### PR TITLE
Andeebranch

### PR DESCRIPTION
**Pull request by andeeisaacs**
Completed my assessment! :D Big thanks to Austin for helping me out in the JS questions.

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -3,12 +3,21 @@
...
+//My message to this final assessment: "So long, and thanks for all the fish"
 
+//Creation of function that will take in an array of objects and join them together as a sentence. 
+const describeHooptyFroods = (array) => {
+    // prepare thyself for a super long return. 
+    return  array.reduce( (paragraph, currentPerson) => `${paragraph} "${[ currentPerson['name'].split(' ').map( (name) => { return name[0].toUpperCase() + name.slice(1) }).join(' '), currentPerson['occupation'] ].join( currentPerson['occupation'].split(' ')[0] === 'president' ? ' is the ' : ' is a ')}."`,'')
+}
 
+//Calling our function in a console.log. 
+console.log(describeHooptyFroods(people))
```
Comment (rachaelstanislaw) [inline code-challenges.js:20]: yooo

File: code-challenges.js [hunk 2]
```diff
@@ -18,11 +27,25 @@ var testingArray2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
 
+//Creation of a function that takes in a mixed data array and returns an array of only the remainder of the numbers when divided by 3.
+const mixedFloat = (array) => {
+     return array.filter( value => typeof( value ) === "number" ).map( value => value % 3)
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:33]: v concise v great job

File: code-challenges.js [hunk 3]
```diff
@@ -18,11 +27,25 @@ var testingArray2 = [5, "Hola", 43, -34, "greetings", true]
...
 var testingArray4 = [7, "hi", 3, 1, "hi", 4, "hello", 4, 7]
 // Expected output: [ 3, 7, "hi", 10, "hello", 4, 1 ]
+
+//Creation of function that takes in two arrays as arguments returns one array with no duplicate values.
+const noDupes = ( array1, array2 ) => {
+    return [ ... new Set( array1.concat(array2) ) ]
+}
+
+//Calling our function in a console.log with both arrays in its argument. 
+console.log(noDupes( testingArray3, testingArray4 ))
```
Comment (rachaelstanislaw) [inline code-challenges.js:51]: oh yeeeah

File: interview-questions.md [hunk 4]
```diff
@@ -4,43 +4,79 @@ Answer the following questions. First, without external resources. Challenge you
 1. In a model called Animal that has_many Sightings, what is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
 
   Your answer:
-
+  The foreign_key would live inside the Sightings(belongs_to Animal) model as a column. 
+  The foreign_key would be "animal_id"
```
Comment (rachaelstanislaw) [inline interview-questions.md:8]: yup!

File: interview-questions.md [hunk 5]
```diff
@@ -4,43 +4,79 @@ Answer the following questions. First, without external resources. Challenge you
...
 
 
 2. Which routes must always be passed params and why?
 
   Your answer:
+  Show - Requires an id param to be able to pick out the specific row/entry you are looking for. 
+  Update - Requires the id param to be able to pick out the specific row/entry you are looking to update/modify
+  Delete - Requires the id param to be able to pick out the specific row/entry you are looking to DESTROY (muahaha.)
+  
+  And honestly, if you're doing anything somewhat custom in your controller, you might be needing to create a route that will require a param passed to it, but the shown examples above are just three of the basics I can think of at this time when using something like "generate resource" that creates a set of defaults for you.
```
Comment (rachaelstanislaw) [inline interview-questions.md:20]: mhmm!!

File: interview-questions.md [hunk 6]
```diff
@@ -4,43 +4,79 @@ Answer the following questions. First, without external resources. Challenge you
...
+  This would be a huge wall of text, if I were to copy paste directly in, so I'm not going to do that. However the best resource is directly from Rails guide documentation itself. So heres a link! https://guides.rubyonrails.org/routing.html
 
 
 
 3. Write a rails route for a controller called "main" and a page called "game" that takes in a parameter called "guess".
 
   Your answer:
+  In the situation of a interview, I'd be asking a few more questions- such as what the model name is and what is goin' on in the controller. So, to the best of my abilities I'm working only with what I have here. 
+get '/game/:guess', to: 'main#game'
+
```
Comment (rachaelstanislaw) [inline interview-questions.md:32]: looks good to me! great thinking about what further questions you'd ask in an interview

File: interview-questions.md [hunk 7]
```diff
@@ -4,43 +4,79 @@ Answer the following questions. First, without external resources. Challenge you
...
+  -A model (table)
+  -A migration file
+  -A view folder 
+  -A routes file (with some default routes set up, such as index and show, etc)
+  -A controller
+  -A scss file
+  -A helper folder
+  -An asset folder. 
+  -Much more! :O
+
```
Comment (rachaelstanislaw) [inline interview-questions.md:56]: boom!

File: interview-questions.md [hunk 8]
```diff
@@ -4,43 +4,79 @@ Answer the following questions. First, without external resources. Challenge you
...
+/users/       method="POST"   
+- This is the magic. REAL MAGIC. It can coincide with the above route (new) that will take in a new entry of data and POST it to our table (users) as a new row. 
+- In other words, POST is responsible for the creation of new rows to a specific model. 
 
-/users/1      method="PUT"    
+/users/1      method="PUT"
+-This is our UPDATE with a specific param (1) passed to it that will allow us to update/patch our specific row/entry with modified information. 
 
 /users/1      method="DELETE"
+-This is our DESTROY (.. muahah) with a specific param (1) passed to it that will allow us to identify which row of the model/table we are looking to delete/destroy.
```
Comment (rachaelstanislaw) [inline interview-questions.md:82]: yay!!

**Review by rachaelstanislaw — APPROVED** [review]
excellent work! go ahead and merge :)

**Comment by andeeisaacs** [comment]
Thank you so much Rachael!

**Pull request merged**